src/server/scripts/EasternKingdoms/zone_silverpine_forest.cpp

```diff
@@ -1596,6 +1618,654 @@ struct npc_silverpine_forsaken_bat : public VehicleAI
     ObjectGuid _playerGUID;
 };
 
+Position const LordaeronCancelScenePos = { 498.54f, 1560.2840f, 128.2032f, 1.100281f };
+
+enum ForsakenWarhorsePlayer
+{
+    NPC_FORSAKEN_WARHORSE_SYLVANAS              = 45057,
+
+    SPELL_RIDE_SYLVANAS_HORSE                   = 84166,
+    SPELL_LORDAERON_COMPLETE                    = 84185,
+    SPELL_DESPAWN_ALL_SUMMONS_LORDAERON         = 84173,
+    SPELL_FADE_TO_BLACK                         = 89092,
+
+    EVENT_ACTIVATE_SKIP                         = 1
+};
+
+// 45041 - Forsaken Warhorse (Player)
+struct npc_silverpine_warhorse_player_lordaeron : public ScriptedAI
+{
+    npc_silverpine_warhorse_player_lordaeron(Creature* creature) : ScriptedAI(creature) { }
+
+    void JustAppeared() override
+    {
+        me->HandleEmoteCommand(EMOTE_ONESHOT_MOUNT_SPECIAL);
+
+        me->CastSpell(nullptr, SPELL_RIDE_SYLVANAS_HORSE, true);
+
+        me->SetUnitFlag(UNIT_FLAG_IMMUNE_TO_PC);
+        me->SetUnitFlag(UNIT_FLAG_IMMUNE_TO_NPC);
+
+        me->SetReactState(REACT_PASSIVE);
+    }
+
+    void IsSummonedBy(WorldObject* summoner) override
+    {
+        _playerGUID = summoner->GetGUID();
+    }
+
+    void EnterEvadeMode(EvadeReason /*why*/) override { }
+
+    void UpdateAI(uint32 diff) override
+    {
+        _events.Update(diff);
+
+        while (uint32 eventId = _events.ExecuteEvent())
+        {
+            switch (eventId)
+            {
+                case EVENT_ACTIVATE_SKIP:
+                {
+                    if (Player* player = ObjectAccessor::GetPlayer(*me, _playerGUID))
```

Comment:
just use `if (Unit* summoner = me->GetSummonerUnit()` its doing all the checks by itself, you can drop `_playerGUID` in the process

Reply:
Feedback applied though I need to access ToTempSummon() for this.